core/src/main/scala/org/apache/spark/sql/execution/row/RowFormatRelation.scala

```diff
@@ -64,28 +67,65 @@ class RowFormatRelation(
 
   override def toString: String = s"RowFormatRelation[$table]"
 
-  lazy val connectionType = ExternalStoreUtils.getConnectionType(dialect)
+  protected val connectionType = ExternalStoreUtils.getConnectionType(dialect)
 
   final lazy val putStr = ExternalStoreUtils.getPutString(table, schema)
 
+  private[this] lazy val resolvedName = ExternalStoreUtils.lookupName(table,
+    tableSchema)
+
+  @transient private[this] lazy val region: LocalRegion =
+    Misc.getRegionForTable(resolvedName, true).asInstanceOf[LocalRegion]
+
+  private[this] lazy val indexedColumns: mutable.HashSet[String] = {
+    val cols = new mutable.HashSet[String]()
+    val im = region.getIndexUpdater.asInstanceOf[GfxdIndexManager]
+    if (im != null && im.getIndexConglomerateDescriptors != null) {
+      val baseColumns = im.getContainer.getTableDescriptor.getColumnNamesArray
+      val itr = im.getIndexConglomerateDescriptors.iterator()
+      while (itr.hasNext) {
+        // first column of index has to be present in filter to be usable
+        val indexCols = itr.next().getIndexDescriptor.baseColumnPositions()
+        cols += baseColumns(indexCols(0))
+      }
+    }
+    cols
+  }
+
+  override def unhandledFilters(filters: Array[Filter]): Array[Filter] =
```

Comment:
We don't handle unHandledPredicates in StoreDataSourceStrategy. Can be another follow up check in to this one. Will track that.